presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java

```diff
@@ -233,5 +243,31 @@ public String toString()
                     .add("index", index)
                     .toString();
         }
+
+        private static final class MaskingBlockLoader
+                implements LazyBlockLoader
+        {
+            private Function<Block, Block> maskDeletedRowsFunction;
+            private Block sourceBlock;
+
+            public MaskingBlockLoader(Function<Block, Block> maskDeletedRowsFunction, Block sourceBlock)
+            {
+                this.maskDeletedRowsFunction = maskDeletedRowsFunction;
+                this.sourceBlock = sourceBlock;
+            }
+
+            @Override
+            public Block load()
+            {
+                checkState(maskDeletedRowsFunction != null, "Already loaded");
+
+                Block resultBlock = maskDeletedRowsFunction.apply(sourceBlock);
```

Comment:
`sourceBlock` -> `sourceBlock.getLoadedBlock()`

Reply:
getters on sourceBlock will load it when reading. Will getLoadedBlock help in any other way?